server/app/mutations/host_nodes/evacuate.rb

```diff
@@ -0,0 +1,50 @@
+require_relative 'common'
+
+module HostNodes
+  class Evacuate < Mutations::Command
+    include Workers
+
+    required do
+      model :host_node
+    end
+
+    def validate
+      add_error(:host_node, :already_evacuated, "Node #{self.host_node.name} is already evacuated") if self.host_node.evacuated?
+    end
+
+    def execute
+      self.host_node.set(:evacuated => true)
+
+      re_deploy_needed_services(self.host_node)
```

Comment:
`GridScheduler.reschedule` would run for all services. In evacuation case only some of the services need scheduling.